### DISCUSSION TITLE
Validating `Json` field against a schema (v.2.0)

### DISCUSSION THREAD

**Discussion by treylav** [pydantic/pydantic#6354] [Question] [answered]
Hello.

I have a JSON type field and an external JSON schema (for example, very simplified):
```python
import pydantic

schema: dict = {
	"type": "object",
	"properties": {
		"avatar": {
			"type": "array",
			"maxItems": 4,
			"items": {
				"type": "integer"
			}
		},
		"required": [
			"avatar"
		]
	}
}


class Dimensions(pydantic.BaseModel):
    value: pydantic.Json
```

How do I check if a passed value matches a schema in Pydantic 2.0? I can, of course, install a schema validation package and add a validator with `mode="before"`, but I'm pretty sure that's not the right way.

Should I use `pydantic.TypeAdapter[pydantic.Json]` for this case?

Thanks in advance.

**Comment by adriangb** [comment, marked as answer]
See https://docs.pydantic.dev/latest/usage/types/json/. You need to create a type (`BaseModel`, `TypedDict`, etc.) and use it like `field: Json[MyType]`. Pydantic will first load and validate the json string.
